site/src/pages/TemplatePage/TemplateInsightsPage/IntervalMenu.tsx

```diff
@@ -0,0 +1,87 @@
+import CheckOutlined from "@mui/icons-material/CheckOutlined";
+import ExpandMoreOutlined from "@mui/icons-material/ExpandMoreOutlined";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
+import { useState, useRef } from "react";
+
+export const insightsIntervals = {
+  day: {
+    label: "Daily",
+  },
+  week: {
+    label: "Weekly",
+  },
+} as const;
+
+export type InsightsInterval = keyof typeof insightsIntervals;
+
+export const IntervalMenu = ({
+  value,
+  onChange,
+}: {
+  value: InsightsInterval;
+  onChange: (value: InsightsInterval) => void;
+}) => {
+  const anchorRef = useRef<HTMLButtonElement>(null);
+  const [open, setOpen] = useState(false);
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <div>
+      <Button
+        ref={anchorRef}
+        id="interval-button"
```

Comment:
Could the hard-coded ID value be swapped out for a value derived from React's `useId`?

Reply:
We can but what value do you see on that?

Reply:
So after double-checking, I don't think that there's much risk with keeping it hard-coded, since I'm guessing the component will only ever be used in one spot

I guess it's more out of habit that I use it for every component, but it does serve a purpose for more reusable ones, especially if multiple of them might be on the screen at one time

Reply:
Yes, right now the idea is not making it reusable.